### PR TITLE
Adds a Heresy phobia, fixes some typos in heretic code, places the stargazer and fire shark under heretic_summon

### PR DESCRIPTION
**Pull request by DATA-xPUNGED**
## About The Pull Request

Instead of a supernatural phobia, sacrificed people will now get a HERESY phobia, which specifically targets heretic stuff, instead of all of magic. Other heretics, however, do not get this, but instead get knowledge points sapped from them, as their mind is ruined in a different way than normal people. This comes with different flavour text and a harsher but shorter lasting mood debuff. Also fixes some "knowlege" code typos and places the star gazer and fire shark all in under heretic_summon
## Why It's Good For The Game

Heretics, Wizards, Chaplains and Cultists will now be able to keep playing the game after getting sacrificed without overdosing on psicodine, while still retaining the phobia's original design purpose. placing the fire shark and star gazer under the same type also is useful as they share a lot of values. typos bad.
## Changelog
:cl:
add: Sacrificed heretic targets will now receive a phobia of heresy instead of a phobia of the supernatural. Sacrificed heretics will not get this phobia, but will lose knowledge points instead.
/:cl:
Huge thanks to Helg2 for doing the original PR and helping me with this one, couldn't be done without them

### CODE REVIEW

File: code/_globalvars/phobias.dm [hunk 1]
```diff
@@ -491,6 +523,7 @@ GLOBAL_LIST_INIT(phobia_species, list(
 		/datum/species/abductor,
 		/datum/species/lizard,
 	)),
+
```
Comment (ZephyrTFA) [inline code/_globalvars/phobias.dm:526]:
```suggestion
```

**Review by ZephyrTFA — CHANGES_REQUESTED** [review]

File: code/_globalvars/phobias.dm [hunk 2]
```diff
@@ -68,6 +70,10 @@ GLOBAL_LIST_INIT(phobia_mobs, list(
 		/mob/living/basic/pet/penguin,
 	)),
 	"doctors" = typecacheof(list(/mob/living/simple_animal/bot/medbot)),
+	"heresy" = typecacheof(list(
+		/mob/living/simple_animal/hostile/heretic_summon,
+		/mob/living/basic/heretic_summon,
```
Comment (Helg2) [inline code/_globalvars/phobias.dm:75]:
```suggestion
		/mob/living/basic/heretic_summon,
		/mob/living/simple_animal/hostile/heretic_summon,
```
so it's alphabetical

**Comment by newfren** [comment]
what happens if you've spent all your points and get sacced, do you lose unlocked research?

**Comment by Jacquerel** [comment]
I think that giving the basic heretic mobs a shared base type is a good idea but collapsing them into one file less so, we don't necessarily *want* all these mobs in one file and someone else will just separate them again later. One folder would be more sensible.

File: code/modules/mob/living/basic/heretic/fire_shark.dm [hunk 3]
```diff
@@ -29,11 +23,11 @@
 	death_message = "implodes into itself."
 	ai_controller = null
```
Comment (Helg2) [inline code/modules/mob/living/basic/heretic/fire_shark.dm:24]:
```suggestion
```
I think you can delete it because the `heretic_summon` already has it

**Comment by DATA-xPUNGED** [comment]
After asking for help and getting from 4 different people i finally did it. No unit tests to scream at me anymore

File: code/_globalvars/phobias.dm [hunk 4]
```diff
@@ -356,6 +363,31 @@ GLOBAL_LIST_INIT(phobia_objs, list(
...
+		/obj/item/melee/rune_carver,
+		/obj/item/melee/sickly_blade,
+		/obj/item/melee/touch_attack/mansus_fist,
+		/obj/item/toy/eldritch_book,
+		/obj/item/toy/reality_pierce,
+		/obj/structure/destructible/eldritch_crucible,
+		/obj/projectile/curse_hand,
+		/obj/item/gun/ballistic/rifle/lionhunter,
+		/obj/item/ammo_box/a762/lionhunter,
+	)),
```
Comment (Helg2) [inline code/_globalvars/phobias.dm:390]:
```suggestion
	"heresy" = typecacheof(list(
		/obj/effect/floating_blade,
		/obj/effect/forcefield/cosmic_field,
		/obj/effect/heretic_rune,
		/obj/effect/heretic_influence,
		/obj/effect/visible_heretic_influence,
		/obj/item/ammo_box/a762/lionhunter,
		/obj/item/clothing/mask/madness_mask,
		/obj/item/clothing/neck/eldritch_amulet,
		/obj/item/clothing/neck/fake_heretic_amulet,
		/obj/item/clothing/neck/heretic_focus,
		/obj/item/clothing/suit/hooded/cultrobes/eldritch,
		/obj/item/codex_cicatrix,
		/obj/item/gun/ballistic/rifle/lionhunter,
		/obj/item/melee/rune_carver,
		/obj/item/melee/sickly_blade,
		/obj/item/melee/touch_attack/mansus_fist,
		/obj/item/reagent_containers/cup/beaker/eldritch,
		/obj/item/toy/eldritch_book,
		/obj/item/toy/reality_pierce,
		/obj/projectile/curse_hand,
		/obj/structure/destructible/eldritch_crucible,
	)),
```
Alphabetize + Deleting cult runes, why are they even here + deleted second eldritch_book

**Comment by DATA-xPUNGED** [comment]
> Also i think you can safely remove `/obj/effect/heretic_influence` cause it won't affect anything, they are invisible and the ones that appear later are a different object

Ideally that will never happen on normal gameplay, but i'd rather keep it there if some forsaken admemery makes that happen

File: code/_globalvars/phobias.dm [hunk 5]
```diff
@@ -356,6 +363,30 @@ GLOBAL_LIST_INIT(phobia_objs, list(
...
+		/obj/effect/visible_heretic_influence,
+		/obj/item/ammo_box/strilka310/lionhunter,
+		/obj/item/ammo_casing/strilka310/lionhunter,
+		/obj/item/clothing/mask/madness_mask,
+		/obj/item/clothing/neck/eldritch_amulet,
+		/obj/item/clothing/neck/fake_heretic_amulet,
+		/obj/item/clothing/neck/heretic_focus,
+		/obj/item/clothing/suit/hooded/cultrobes/eldritch,
+		/obj/item/codex_cicatrix,
+		/obj/item/gun/ballistic/rifle/lionhunter,
```
Comment (Helg2) [inline code/_globalvars/phobias.dm:380]:
```suggestion
		/obj/item/coin/eldritch,
		/obj/item/gun/ballistic/rifle/lionhunter,
```
missed heretic item

File: strings/phobia.json [hunk 6]
```diff
@@ -263,6 +263,38 @@
...
+		"flesh",
+		"ash",
+		"void",
+		"blade",
+		"cosmic",
+		"space",
+		"star",
+		"armsy",
+		"worm",
+		"flesh snek",
```
Comment (Ghommie) [inline strings/phobia.json:284]: "flesh" without "snek" is already in this same list.

File: strings/phobia.json [hunk 7]
```diff
@@ -263,6 +263,38 @@
 		"wt550"
 	],
 
+	"heresy": [
+		"eldritch",
+		"ritual",
+		"rune",
+		"focus",
+		"codex cicatrix",
+		"heretic",
```
Comment (Ghommie) [inline strings/phobia.json:272]: I see a "heretic" but no "heresy" in this list.

File: strings/phobia.json [hunk 8]
```diff
@@ -263,6 +263,38 @@
...
+		"stalker",
+		"fire shark",
+		"ghoul",
+		"pierced reality",
+		"reality pierce",
+		"reality crack",
+		"maid in the mirror",
+		"hand",
+		"arm",
+		"void storm"
```
Comment (Ghommie) [inline strings/phobia.json:295]: "void" without "storm" is already in this same list.

**Review by Ghommie — APPROVED** [review]

File: strings/phobia.json (hunk 8 above)
Comment (DATA-xPUNGED) [reply strings/phobia.json:295]: It's tangential but does the term "hand" also triggers with "hands"?

File: strings/phobia.json [hunk 9]
```diff
@@ -263,6 +263,44 @@
...
+		"lionhunter",
+		"maid in the mirror",
+		"mansus",
+		"offer",
+		"pierced reality",
+		"raw prophet",
+		"reality crack",
+		"reality pierce",
+		"ritual",
+		"ritual",
```
Comment (Helg2) [inline strings/phobia.json:292]:
```suggestion
```
Redundant as it's already here

File: code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm [hunk 10]
```diff
@@ -374,8 +373,11 @@
 	sac_target.remove_status_effect(/datum/status_effect/unholy_determination)
 	sac_target.reagents?.del_reagent(/datum/reagent/inverse/helgrasp/heretic)
 	sac_target.clear_mood_event("shadow_realm")
-	sac_target.gain_trauma(/datum/brain_trauma/mild/phobia/supernatural, TRAUMA_RESILIENCE_MAGIC)
-
+	if(IS_HERETIC(sac_target))
+		var/datum/antagonist/heretic/victim_heretic = sac_target.mind?.has_antag_datum(/datum/antagonist/heretic)
+		victim_heretic.knowledge_points -= 3
```
Comment (DATA-xPUNGED) [inline code/modules/antagonists/heretic/knowledge/sacrifice_knowledge/sacrifice_knowledge.dm:378]: i feel like i SHOULD make this into some kind of var but i don't know where i'd place this var

**Pull request merged**